### PR TITLE
Bug 1817524 - Do not show homepage if last tab of section is closed while not in view

### PR DESCRIPTION
**Pull request by HarrisonOg**
Resubmitting this PR after fixing some of the tests that it broke.

Original PR: https://github.com/mozilla-mobile/firefox-android/pull/876

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1817524

### CODE REVIEW

**Comment by HarrisonOg** [comment]
> Could you post the original PR in the description of this one

added the original PR in the description

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt [hunk 1]
```diff
@@ -274,7 +274,8 @@ class DefaultTabsTrayController(
 
         tab?.let {
             val isLastTab = browserStore.state.getNormalOrPrivateTabs(it.content.private).size == 1
-            if (!isLastTab) {
+            val isCurrentTab = browserStore.state.selectedTabId.equals(tabId)
+            if (!isLastTab || !isCurrentTab) {
                 tabsUseCases.removeTab(tabId)
                 showUndoSnackbarForTab(it.content.private)
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt:280]: is this section any different than what was landed in the original PR?

File: fenix/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt [hunk 2]
```diff
@@ -865,6 +868,36 @@ class DefaultTabsTrayControllerTest {
         verify { controller.handleNavigateToBrowser() }
     }
 
+    @Test
+    fun `GIVEN a normal tab is selected WHEN the last private tab is deleted THEN that private tab is removed and an undo snackbar is shown and original normal tab is still displayed`() {
+        val currentTab = TabSessionState(content = ContentState(url = "https://simulate.com", private = false), id = "currentTab")
+        val privateTab = TabSessionState(content = ContentState(url = "https://mozilla.com", private = true), id = "privateTab")
```
Comment (MozillaNoah) [inline fenix/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt:874]: this looks identical as well. (just making sure this is the intent)

**Review by MozillaNoah — APPROVED** [review]
This LGTM. Were these tests silently failing, like the one you brought to my attention last week?

File: fenix/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt (hunk 2 above)
Comment (HarrisonOg) [reply fenix/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt:874]: yeah this test and the change to `TabsTrayController` are staying the same as the original PR. 
the changes are to two other tests that were silently failing for me but were revealed last week.

**Pull request merged**